src/cpp/fastdds/publisher/qos/DataWriterQos.cpp

```diff
@@ -60,9 +60,9 @@ WriterQos DataWriterQos::get_writerqos(
     {
         uint64_t id = 0;
         Host::uint48 mac_id = Host::instance().mac_id();
-        for (size_t i = 0; i < Host::mac_id_length; ++i)
+        for (size_t i = 1; i < Host::mac_id_length; ++i)
```

Comment:
Looking at this, I think the code was completely wrong in the first place. The intention here was to convert the 6-octet array on `mac_id` to a `uint64_t` with the lowest 16-bits having a value of 0.

```suggestion
        for (size_t i = 0; i < Host::mac_id_length; ++i)
```